lib/Runtime/Language/DynamicProfileInfo.cpp

```diff
@@ -1921,7 +1921,7 @@ namespace Js
 
             if (callSiteInfoCount != 0)
             {
-                callSiteInfo = RecyclerNewArrayLeaf(recycler, CallSiteInfo, callSiteInfoCount);
+                callSiteInfo = RecyclerNewArray(recycler, CallSiteInfo, callSiteInfoCount);
```

Comment:
After more look and discussion offline, I'll revert this change and add a comment. the Leaf here is safe because we explicitly save the only pointer elsewhere.